lib/vector-common/src/internal_event/optional_tag.rs

```diff
@@ -0,0 +1,14 @@
+/// The user can configure whether a tag should be emitted. If they configure it to
+/// be emitted, but the value doesn't exist - we should emit the tag but with a value
+/// of `-`.
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub enum OptionalTag {
+    Ignored,
+    Specified(Option<String>),
```

Comment:
The Source comes from the event metadata, and is stored as an `Arc<ComponentKey>`, so I am able to just clone that. The service, however is a field on the event - so will be a `Value`. There isn't really anyway I can easily keep a reference to this that I can see.

Reply:
Since all events in a batch would share the same service (I think?) would it make sense to create an `Arc` before creating the batch and then clone that into each event?